devdoc/two_dimensional_array_requirements.md

```diff
@@ -0,0 +1,106 @@
+# two_dimensional_array_requirements
+============================
+
+## Overview
+
+`two_dimensional_array` provides a 2D array with variable size of rows and columns. Each row has the same size to ensure easier access. The first dimention of array is resizable, it contains the pointer to the second dimention of array. The pointer is initiazlied to NULL and points to the allocated memory address when there's not enough space. When the memory stored in a row is not needed any more, the entire row shall be freed and set back to `NULL`.
```

Comment:
Can we have some CRLFs so we break the lines so that we can read the text better?